state/protocol/badger/state.go

```diff
@@ -42,17 +43,19 @@ func Bootstrap(
 	commits storage.EpochCommits,
 	statuses storage.EpochStatuses,
 	root protocol.Snapshot,
+	skipNwAddressBasedValidations bool,
```

Comment:
A few thoughts:

1. We are using this config flag during the bootstrap, then also storing it in the `State` struct. As far as I can see that stored value is never used (eg [in `mutator.go`](https://github.com/onflow/flow-go/pull/1266/files#diff-c79d3fc5d32f153917b7cd923ab4ba64ef22f12268ddd13f353a06391f68a96dR817) we pass in `false` rather than use the stored flag. I would suggest we stop storing this value in the struct if we aren't using it, and only modify the validation logic at startup, when we are validating the stripped snapshot.

2. I'd prefer we avoid the pattern of adding configs as required parameters, especially for things like this where the config represents an uncommon edge case. It makes the common case of calling the function more concise and makes modifications to the common case more explicit.

```go
type BootstrapConfig struct {
  SkipNetworkAddressValidation bool
}

func WithSkipNetworkAddressValidation(conf *BootstrapConfig) {
  conf.SkipNetworkAddressValidation = true
}

// at the majority of callsites - no change
Bootstrap(...)
// where needed
Bootstrap(..., WithSkipNetworkAddressValidation) 
```

3. Rather than passing the config all the way down to the validation function (eg. `isValidEpochSetup`), we could make the validation functions return descriptive sentinel errors for the validation cases we want to treat differently, then handle those sentinels based on the config, in the same layer the config has already been defined (`Bootstrap`).

```go
err := isValidEpochSetup(...)
if errors.Is(err, ErrDuplicateNetworkingAddress) && !SkipNetworkAddressValidation {
  return err
}  else if err != nil {
  return err
}
```